tests/epyccel/test_epyccel_lists.py

```diff
@@ -658,3 +670,23 @@ def g():
     epyc_f = epyccel(f, language=language)
     assert f() == epyc_f()
 
+def test_mutable_indexing(stc_language):
```

Comment:
I don't know if you wanna handle the following case:
```

if __name__ == '__main__':
    x = [1,2,3]
    x[x[0]] = 5
    print(x)
```
which outputs:
```
...
 (*vec_int64_t_at_mut(&x,(*vec_int64_t_at_mut(&x,INT64_C(0))))) = INT64_C(5);
...
```
Not everyting in lhs should be at_mut

Reply:
I agree that this output is not ideal, but I think the generated code should work. I suspect that it is quite difficult to recognise these cases. There is a similar issue with:
```python
x = [[1, 2, 3], [4, 5, 6]]
x[0][0] = 5
```
which should ideally give:
```c
 (*vec_int64_t_at_mut(vec_vec_int64_t_at(&x,INT64_C(0)), INT64_C(0))) = INT64_C(5);
```
but currently gives:
```c
 (*vec_int64_t_at_mut(vec_vec_int64_t_at_mut(&x,INT64_C(0)), INT64_C(0))) = INT64_C(5);
```

I think that as long as the code runs this is sufficient for a first attempt in order to unblock all other developments related to lists. I will open an issue so that we remember that this implementation is sub-optimal

Reply:
Excellent, thank you!

Reply:
See #1955